src/components/tables/DataTable.stories.ts

```diff
@@ -0,0 +1,304 @@
+          <Button icon variant="text" size="sm"
+            ><Icon name="more-fill" color="primary"
+          /></Button>
+        </template>
+      </DataTable>
+    </div>`,
+});
+
+const data = [
+  {
+    id: 1,
+    name: 'Lefteris',
+    title: 'Director of Product',
+    email: 'Lefteris@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 2,
+    name: 'Kelsos',
+    title: 'Director of Product',
+    email: 'Kelsos@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 3,
+    name: 'Yabir',
+    title: 'Director of Product',
+    email: 'Yabir@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 4,
+    name: 'Luki',
+    title: 'Director of Product',
+    email: 'Luki@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 5,
+    name: 'Celina',
+    title: 'Director of Product',
+    email: 'Celina@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 6,
+    name: 'Joseph',
+    title: 'Director of Product',
+    email: 'Joseph@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  {
+    id: 7,
+    name: 'Dimitry',
+    title: 'Director of Product',
+    email: 'Dimitry@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  },
+  ...[...new Array(43)].map((_, index) => ({
+    id: index + 8,
+    name: 'Lindsay Walton',
+    title: 'Front-end Developer',
+    email: 'lindsay.walton@example.com',
+    role: 'Member',
+    date: '10.09.2023',
+  })),
+];
+
+const columns: TableColumn[] = [
+  {
+    key: 'id',
+    label: 'ID',
+  },
+  {
+    key: 'name',
+    label: 'Full name',
+    sortable: true,
+    align: 'right',
+  },
+  {
+    key: 'title',
+    label: 'Job position',
+    sortable: true,
+    align: 'left',
+  },
+  {
+    key: 'email',
+    label: 'Email address',
+    sortable: true,
+  },
+  {
+    key: 'role',
+    sortable: true,
+  },
+  {
+    key: 'action',
+  },
+];
+
+const meta: Meta<Props> = {
+  title: 'Components/Tables/DataTable',
+  component: DataTable,
+  tags: ['autodocs'],
+  render,
+  argTypes: {},
+  args: {
+    modelValue: undefined,
+    rowAttr: 'id',
+    rows: [],
+    columnAttr: 'label',
+    loading: false,
+    dense: false,
+    outlined: false,
+  },
+  parameters: {
+    docs: {
+      controls: {
+        exclude: [
+          'default',
+          'update:model-value',
+          'update:pagination',
+          'update:sort',
+          'tfoot',
+          '`${column.key}-header`',
+          '`${column.key}-data`',
+        ],
+      },
+    },
+  },
+};
+
+type Story = StoryObj<Props>;
+
+export const Default: Story = {
+  args: {
+    rows: data,
+    cols: columns,
+    sort: [{ column: 'name', direction: 'asc' }],
+    pagination: { limit: 10, page: 1, total: 50 },
+  },
+};
+
+export const Dense: Story = {
+  args: {
+    rows: data,
+    dense: true,
+  },
+};
+
+export const Loading: Story = {
+  args: {
+    rows: [],
```

Comment:
rows: data

without this the storybook section become empty

Reply:
thanks, I missed this...

Reply:
I see the issue, when columns is not passed, no data to generate the columns as rows are empty...
I intentionally wanted the table to show only loading state and headers, without body, you can check again.